From 4e310d6d52e629609e87c6293dbd264bfdd37a5b Mon Sep 17 00:00:00 2001
From: "Fan, Yugang" <yugang.fan@intel.com>
Date: Thu, 29 Aug 2019 20:46:51 +0800
Subject: [PATCH] Ignore the EGL_CONTEXT_PRIORITY_LEVEL_IMG attritute.

Ignore when up-level pass EGL_CONTEXT_PRIORITY_LEVEL_IMG
to swiftshader to create egl context. Seems swiftshader can't
support this attribute, we just skip it and continue context
creating.

Change-Id: Ic7d5ecca6967a79d2a6501b15222b70844cb0fbc
Sighed-off-by: Fan, Yugang <yugang.fan@intel.com>
Tracked-On: None
---
 src/OpenGL/libEGL/libEGL.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/OpenGL/libEGL/libEGL.cpp b/src/OpenGL/libEGL/libEGL.cpp
index 5c561b959452..b78d7820b584 100644
--- a/src/OpenGL/libEGL/libEGL.cpp
+++ b/src/OpenGL/libEGL/libEGL.cpp
@@ -830,6 +830,9 @@ EGLContext CreateContext(EGLDisplay dpy, EGLConfig config, EGLContext share_cont
 					return error(EGL_BAD_ATTRIBUTE, EGL_NO_CONTEXT);
 				}
 				break;
+			case EGL_CONTEXT_PRIORITY_LEVEL_IMG:
+				TRACE("Ignore EGL_CONTEXT_PRIORITY_LEVEL_IMG attribute");
+				break;
 			default:
 				return error(EGL_BAD_ATTRIBUTE, EGL_NO_CONTEXT);
 			}
-- 
2.7.4

